recipes/nginx/build.sh

```diff
@@ -0,0 +1,133 @@
+      --http-scgi-temp-path=$PREFIX/var/tmp/nginx/scgi \
+      --http-uwsgi-temp-path=$PREFIX/var/tmp/nginx/uwsgi \
+      --sbin-path=sbin/nginx \
+      --conf-path=$PREFIX/etc/nginx/nginx.conf \
+      --modules-path=lib/nginx/modules \
+      --with-threads \
+      --with-http_ssl_module \
+      --with-http_v2_module \
+      --with-http_realip_module \
+      --with-http_addition_module \
+      --with-http_xslt_module=dynamic \
+      --with-http_sub_module \
+      --with-http_gunzip_module \
+      --with-http_gzip_static_module \
+      --with-http_auth_request_module \
+      --with-http_secure_link_module \
+      --with-http_stub_status_module \
+      --with-stream=dynamic \
+      --with-http_image_filter_module=dynamic \
+      --with-pcre \
+      --with-pcre-jit \
+      --with-cc-opt="$cc_opt" \
+      --with-ld-opt="$ld_opt" \
+      --prefix="$PREFIX"
+
+
+elif [[ $(uname -s) == Linux ]]; then
+  export cc_opt="-I$PREFIX/include -I$PREFIX/include/libxml2 -I$PREFIX/include/libexslt -I$PREFIX/include/libxslt -I$PREFIX/include/openssl"
+  export cc_opt="$cc_opt -O2 -g -pipe -fPIC -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fno-strict-overflow -m64 -mtune=generic -fstack-protector --param=ssp-buffer-size=4"
+  # TODO: for later gcc, change -fstack-protector to -fstack-protector-strong
+  export ld_opt="-L$PREFIX/lib -Wl,-z,relro,-z,now"
+  # http://security.stackexchange.com/questions/24444/what-is-the-most-hardened-set-of-options-for-gcc-compiling-c-c
+  # https://blog.mayflower.de/5800-Hardening-Compiler-Flags-for-NixOS.html
+  # TODO: ASLR
+  ./configure \
+      --http-log-path=var/log/nginx/access.log \
+      --error-log-path=var/log/nginx/error.log \
+      --pid-path=var/run/nginx/nginx.pid \
+      --lock-path=var/run/nginx/nginx.lock \
+      --http-client-body-temp-path=var/tmp/nginx/client \
+      --http-proxy-temp-path=var/tmp/nginx/proxy \
+      --http-fastcgi-temp-path=var/tmp/nginx/fastcgi \
+      --http-scgi-temp-path=var/tmp/nginx/scgi \
+      --http-uwsgi-temp-path=var/tmp/nginx/uwsgi \
+      --sbin-path=sbin/nginx \
+      --conf-path=etc/nginx/nginx.conf \
+      --modules-path=lib/nginx/modules \
+      --with-threads \
+      --with-http_ssl_module \
+      --with-http_v2_module \
+      --with-http_realip_module \
+      --with-http_addition_module \
+      --with-http_sub_module \
+      --with-http_gunzip_module \
+      --with-http_gzip_static_module \
+      --with-http_auth_request_module \
+      --with-http_secure_link_module \
+      --with-http_stub_status_module \
+      --with-http_xslt_module=dynamic \
+      --with-stream=dynamic \
+      --with-http_image_filter_module=dynamic \
+      --with-pcre \
+      --with-pcre-jit \
+      --with-cc-opt="$cc_opt" \
+      --with-ld-opt="$ld_opt" \
+      --prefix="$PREFIX"
+
+      # this is removed for now because libgd apparently needs libwebp, not yet compiled for linux
+
+fi
+
+make -j2
+make install
```

Comment:
Nope, but it has a comprehensive test after it has been compiled, look at the meta.yaml test section.